packages/core/src/store/store.ts

```diff
@@ -610,4 +655,29 @@ export class DefaultAllureStore implements AllureStore, ResultsVisitor {
       ...(this.#environmentsConfig?.[env]?.variables ?? {}),
     };
   }
+
+  dumpState(): AllureStoreDump {
+    return {
+      testResults: mapToObject(this.#testResults),
+      attachments: mapToObject(this.#attachments),
+      testCases: mapToObject(this.#testCases),
+      fixtures: mapToObject(this.#fixtures),
+      environments: this.#environments,
+      reportVariables: this.#reportVariables,
+    };
+  }
+
+  async restoreState(stateDump: AllureStoreDump, attachmentsContents: Record<string, ResultFile> = {}) {
+    const { testResults, attachments, testCases, fixtures, reportVariables, environments } = stateDump;
+
+    mergeMapWithRecord(this.#testResults, testResults);
+    mergeMapWithRecord(this.#attachments, attachments);
+    mergeMapWithRecord(this.#testCases, testCases);
+    mergeMapWithRecord(this.#fixtures, fixtures);
+
+    this.#addEnvironments(environments);
+
+    Object.assign(this.#reportVariables, reportVariables);
+    Object.assign(this.#attachmentContents, attachmentsContents);
```

Comment:
This seems not to be enough for attachments to work. I observed the attachments directory missing in `allure-report/awesome/data`, and the attachments, while being present on the test result pages, appeared empty.

Unfortunately, I can't check this after the rework because the whole report is now not generating properly.